assets/js/util.js

```diff
@@ -242,8 +243,19 @@ function synchronizeTextareaHeights() {
     $('#translatedText').css('height', originalTextScrollHeight + 'px');
 }
 
-/*:: export {synchronizeTextareaHeights, modeEnabled, ajaxSend, ajaxComplete, filterLangList, onlyUnique,
-    SPACE_KEY_CODE, ENTER_KEY_CODE, HTTP_OK_CODE, HTTP_BAD_REQUEST_CODE, XHR_LOADING, XHR_DONE} */
+function callApy(options, endpoint) {
+    return $.ajax(Object.assign({
+        url: config.APY_URL + endpoint,
+        beforeSend: ajaxSend,
+        contentType: 'application/x-www-form-urlencoded; charset=UTF-8',
+        dataType: 'json',
+        type: (options.data.q.length > THRESHOLD_REQUEST_LENGTH) ? 'POST' : 'GET'
```

Comment:
> $.param() might come to rescue!

Looks promising.

> followed by the q= 'encodeddata' which is nothing but the contents of originalText?

There are other parameters. This should be built in a way that is extensible for other modes as well. Don't just think about the translation mode! We want to build things that generalize well.

> And it the length of this URL that I want?

Yes. The limiting factor is the size of the URL.